Feature/metadataformat

xchapter7x opened the pull request:
This adds a model and supporting Load function to allow for reading metadata files.

This applies to the issue here https://github.com/policy-hub/policy-hub-cli/issues/1

xchapter7x committed:
initial commit for metadata config

- stubs out description and version fields
- adds loading function to consume config files

[https://github.com/policy-hub/policy-hub-cli/issues/1]

xchapter7x committed:
add support for multiple package in meta file

- multiple packages should be able to be defined in
  a single meta file

xchapter7x committed:
add support for labels to metadata model

xchapter7x committed:
add support for maintainers to metadata model

xchapter7x committed:
adds path and homepage to metadata model

xchapter7x committed:
add support for image url to metadata model

xchapter7x committed:
adds support for name field in metadata model

Blokje5 commented on pkg/metaconfig/metaconfig.go:
```diff
@@ -0,0 +1,20 @@
+
+import "gopkg.in/yaml.v2"
+
+type MetadataConfig struct {
```
I would argue for just Metadata. MetadataConfig seems to imply this is configuration of Metadata while it is actually just Metadata.

xchapter7x replied:
agreed. i was struggling with this name.

Blokje5 reviewed (changes requested):
Small naming request change, but LGTM!

xchapter7x committed:
struct name change for more clarity

  - addresses feedback here https://github.com/policy-hub/policy-hub-cli/pull/7#pullrequestreview-505013015

xchapter7x commented:
@Blokje5 
Change is in... ready to merge whenever you are :)

xchapter7x committed:
Merge branch 'main' into feature/metadataformat

Blokje5 reviewed (approved):
LGTM!
